var/spack/repos/builtin/packages/intel-parallel-studio/package.py

```diff
@@ -86,7 +86,7 @@ def blas_libs(self):
         # TODO: TBB threading: ['libmkl_tbb_thread', 'libtbb', 'libstdc++']
         mkl_libs = find_libraries(
             mkl_integer + ['libmkl_core'] + mkl_threading,
-            root=join_path(self.prefix.lib, 'intel64'),
+            root=join_path(self.prefix, 'mkl', 'lib', 'intel64'),
```

Comment:
maybe in a separate PR which can be quickly merged. Then you will rebase this one to get rid of this change.